### PR TITLE
Upgrade codebase to support koa 2

### PR DESCRIPTION
**Pull request by kevinrambaud**
I recently needed and used this feature in some projects, that's a good opportunity to make this middleware support the Koa of today.

This PR includes :

* added editorconfig
* restricted package usage to `node >= 7.6.0`
* updated devDependencies
* updated code base to use async/await
* updated examples
* updated readme
* updated package version due to breaking change from `0.0.1` to `1.0.0`

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,55 +1,60 @@
...
 ### Object responses
 
-  Script:
+Script:
 
 ```js
-var filter = require('koa-filter');
-var koa = require('koa');
+const Koa = require('koa');
+const filter = require('koa-filter');
```
Comment (coderhaoxin) [inline Readme.md]: should be `koa-json-filter`

**Comment by coderhaoxin** [comment]
LGTM, but I don't have npm rights :)

cc @tj

File: Readme.md (hunk 1 above)
Comment (kevinrambaud) [reply Readme.md]: good catch, just fixed it

**Comment by kevinrambaud** [comment]
Got it @coderhaoxin, will wait for @tj. In the meantime, I found [koa-json-mask](https://github.com/nemtsov/koa-json-mask) package that does the same thing with advanced filtering capabilities.

**Comment by coderhaoxin** [comment]
land 9c90e9453cece7aa4646641c6d50b5272cffa6f5

**Comment by coderhaoxin** [comment]
@kevinrambaud published as `@koa/json-filter` 1.0.0 :)

**Comment by kevinrambaud** [comment]
woo great @coderhaoxin I completely forgot this PR

**Pull request closed without merging**